st2common/st2common/services/datastore.py

```diff
@@ -211,7 +214,7 @@ def _get_api_client(self):
 
         if not self._client or token_expire:
             self._logger.audit('Creating new Client object.')
-            ttl = (24 * 60 * 60)
+            ttl = cfg.CONF.auth.token_ttl
             self._token_expire = get_datetime_utc_now() + timedelta(seconds=ttl)
             temporary_token = create_token(username=self._api_username, ttl=ttl)
```

Comment:
Just adding `service=True` argument to `create_token` function call should do the trick. Then we can revert the TTL change.

Recently I made a change so we support arbitrary TTLs for service tokens - https://github.com/StackStorm/st2/pull/3315